web/src/util/shape-math.test.ts

```diff
@@ -1,6 +1,26 @@
-import { constrainBoxTo } from "./shape-math";
+import { constrainBoxTo, contains } from "./shape-math";
 
-describe("constrainTo", () => {
+describe("contains", () => {
+  it("returns true for points inside the box", () => {
+    expect(
+      contains({ top: 0, left: 0, bottom: 10, right: 10 }, { x: 5, y: 5 })
+    ).toBe(true);
+  });
+
+  it("returns true for points on the edge of the box", () => {
+    expect(
+      contains({ top: 0, left: 0, bottom: 10, right: 10 }, { x: 0, y: 0 })
+    ).toBe(true);
+  });
+
+  it("returns false for points outside the box", () => {
+    expect(
+      contains({ top: 0, left: 0, bottom: 10, right: 10 }, { x: 0, y: 0 })
+    ).toBe(true);
```

Comment:
This test will pass but does not do what it says it does.

Reply:
Whoops, good catch

Reply:
Okay, I fixed this